### PR TITLE
fix(codeserver): don't try to mount a volume if it's target on a specific container

### PR DESCRIPTION
**Pull request by stavros-k**
**Description**

⚒️ Fixes  https://github.com/truecharts/charts/issues/8694

When there is a `targetSelector` defined, don't try to mount this volume to codeserver. As it's most likely a configmap/secret or a volume only relevant to a specific container

eg:
```yaml
persistence:
  init:
    enabled: true
    type: configmap
    objectName: init
    mountPath: "/config/init"
    defaultMode: "0777"
    readOnly: true
    targetSelector:
      main:
        init: {}
```
But the caveat is that also this wont be mounted:
```yaml
persistence:
  vol:
    enabled: true
    type: emptyDir
    mountPath: "/config/init"
    targetSelector:
      main:
        main: {}
```

TL;DR only volumes without a selector or a selector with `targetSelectAll` will be mounted to codeserver.

Not sure what we could do on scale GUI to make this more flexible. Needs more thought.
But this is a quick fix for now.

**⚙️ Type of change**

- [ ] ⚙️ Feature/App addition
- [x] 🪛 Bugfix
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🔃 Refactor of current code

**🧪 How Has This Been Tested?**

**📃 Notes:**

**✔️ Checklist:**

- [x] ⚖️ My code follows the style guidelines of this project
- [x] 👀 I have performed a self-review of my own code
- [ ] #️⃣ I have commented my code, particularly in hard-to-understand areas
- [ ] 📄 I have made corresponding changes to the documentation
- [ ] ⚠️ My changes generate no new warnings
- [ ] 🧪 I have added tests to this description that prove my fix is effective or that my feature works
- [x] ⬆️ I increased versions for any altered app according to semantic versioning

**➕ App addition**

If this PR is an app addition please make sure you have done the following.

- [ ] 🪞 I have opened a PR on [truecharts/containers](https://github.com/truecharts/containers) adding the container to TrueCharts mirror repo.
- [ ] 🖼️ I have added an icon in the Chart's root directory called `icon.png`

---

_Please don't blindly check all the boxes. Read them and only check those that apply.
Those checkboxes are there for the reviewer to see what is this all about and
the status of this PR with a quick glance._

### CODE REVIEW

File: library/common/templates/lib/container/_volumeMounts.tpl [hunk 1]
```diff
@@ -99,7 +99,7 @@ objectData: The object data to be used to render the container.
       {{- end -}}
 
       {{/* If container is selected */}}
-      {{- if or (mustHas $objectData.shortName ($selectorValues | keys)) (eq $objectData.shortName "codeserver") -}}
+      {{- if or (mustHas $objectData.shortName ($selectorValues | keys)) -}}
         {{/* Merge with values that might be set for the specific container */}}
         {{- $volMount = mustMergeOverwrite $volMount (get $selectorValues $objectData.shortName) -}}
```
Comment (stavros-k) [inline library/common/templates/lib/container/_volumeMounts.tpl:104]: Here we are getting the values (eg mountPath) from the selector. but codeserver is not in this selector, making it fail.

We could check if the `get` returns any data and only then use it to "bypass" the error and let codeserver use the default `mountPath` defined outside of the selector (if not defined it will most likely fail later).

But considering that 90% of apps that have actually targetSelector defined, are only used for configmaps, I don't see the reason to mount them to codeserver.

**Comment by Ornias1993** [comment]
Agreed.
Can we also just not mount configmsps and secretsto code server at-all? As those are not editableanyway.

**Comment by stavros-k** [comment]
Updated, now it only ingores the configmap/secret, but it will mount all volumes even if only targeted on a specific container.
In this case, for mountPath it will use the the first's selected container values and if there isn't a specific mountPath there, it fallback to the top level mountPath.
